core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/TableStatistics.scala

```diff
@@ -0,0 +1,139 @@
+package edu.uci.ics.amber.core.storage.result.iceberg
+
+import edu.uci.ics.amber.util.IcebergUtil
+import org.apache.iceberg.Table
+import org.apache.iceberg.types.{Conversions, Types}
+
+import java.nio.ByteBuffer
+import java.time.{Instant, LocalDate, ZoneOffset}
+import java.time.format.DateTimeFormatter
+import scala.jdk.CollectionConverters._
+import scala.collection.mutable
+
+/**
+  * `TableStatistics` provides methods for extracting metadata statistics for the results
+  *
+  * **Statistics Computed:**
+  * - **Numeric fields (Int, Long, Double)**: Computes `min` and `max`.
+  * - **Date fields (Timestamp)**: Computes `min` and `max` (converted to `LocalDate`).
+  * - **All fields**: Computes `not_null_count` (number of non-null values).
+  */
+case class TableStatistics(table: Table) {
+
+  /**
+    * Computes metadata statistics for all fields in the Iceberg table.
+    *
+    * @param table The Iceberg table to analyze.
+    *
+    * @return A `Map` where keys are field names and values are statistics (`min`, `max`, `not_null_count`).
+    */
+  def getTableStatistics: Map[String, Map[String, Any]] = {
```

Comment:
Rename it to `getCombinedStatsMap`, as this method is merging all three types of stats map.